toolchain/lower/file_context.h

```diff
@@ -165,6 +184,69 @@ class FileContext {
+  // by one while lowering their definitions.
+  auto AddLoweredSpecificForGeneric(SemIR::GenericId generic_id,
+                                    SemIR::SpecificId specific_id) {
+    lowered_specifics_[generic_id.index].push_back(specific_id);
+  }
+
+  // Initialize a return a SpecificFunctionFingerprint* instance for specific.
+  // The internal of the fingerprint are populated during and after lowering
+  // the function body of that specific.
+  auto InitializeFingerprintForSpecific(SemIR::SpecificId specific_id)
+      -> SpecificFunctionFingerprint* {
+    if (!specific_id.has_value()) {
+      return nullptr;
+    }
+    return &lowered_specific_fingerprint_[specific_id.index];
+  }
+
+  // Entry point for coalescing equivalent specifics. Two function definitions,
+  // from the same generic, with different specific_ids are considered
+  // equivalent if, at the LLVM level, one can be replaced with the other, with
+  // no change in behavior. All LLVM types and instructions must be equivalent.
+  auto CoalesceEquivalentSpecifics() -> void;
+
+  // While coalescing specifics, compare the function types for two specifics.
+  // This uses a fingerprint generated for each function type.
+  auto AreFunctionTypesEquivalent(SemIR::SpecificId specific_id1,
+                                  SemIR::SpecificId specific_id2) -> bool;
+
+  // While coalescing specifics, compare the function bodies for two specifics.
+  // This uses fingerprints generated during lowering of the function body.
+  auto AreFunctionBodiesEquivalent(
+      SemIR::SpecificId specific_id1, SemIR::SpecificId specific_id2,
+      Set<std::pair<SemIR::SpecificId, SemIR::SpecificId>>&
+          visited_equivalent_specifics) -> bool;
+
+  // When an equivalence is found, update the canonical specific to use for
+  // each of the two Specifics found to be equivalent, replace all uses of one
+  // specific with the canonical one; mark one specific for deletion using
+  // specifics_to_delete.
+  auto ProcessSpecificEquivalence(
+      std::pair<SemIR::SpecificId, SemIR::SpecificId> pair,
+      llvm::SmallVector<SemIR::SpecificId>& specifics_to_delete) -> void;
+
+  // Return whether two specifics have been found to be equivalent.
+  auto IsKnownEquivalence(SemIR::SpecificId specific_id1,
+                          SemIR::SpecificId specific_id2) -> bool;
+
+  // Helper function for inserting a pair into a set of pairs in canonical
+  // form. Also implicitly checks entry already existed if it cannot be
+  // inserted.
+  auto PairInserted(
+      SemIR::SpecificId specific_id1, SemIR::SpecificId specific_id2,
+      Set<std::pair<SemIR::SpecificId, SemIR::SpecificId>>& set_of_pairs)
+      -> bool;
+
+  // Helper function for checking if a pair exists into a set of pairs in
+  // canonical form.
+  auto PairExists(
+      SemIR::SpecificId specific_id1, SemIR::SpecificId specific_id2,
+      Set<std::pair<SemIR::SpecificId, SemIR::SpecificId>> set_of_pairs)
```

Comment:
```suggestion
      const Set<std::pair<SemIR::SpecificId, SemIR::SpecificId>>& set_of_pairs)
```

I assume this was an accidental copy